Ctl 017 add search feature

heupotato opened the pull request.

heupotato committed:
Update dialog_action_item.dart

Change navigator

heupotato committed:
Add search feature to contact list screen

lngyeen commented on lib/ui/screens/contact_list_screen.dart:
```diff
@@ -95,6 +99,15 @@ class _ContactListScreenState extends State<ContactListScreen> {
 
+  _onChanged(String value){
+    setState(() {
+      List<Contact> contactList = ContactsRepository.getAllContacts()!;
```
ContactsRepository.getAllContacts() should always return List<Contact>, not optional value.

lngyeen commented on lib/ui/screens/contact_list_screen.dart:
```diff
@@ -19,10 +20,13 @@ class ContactListScreen extends StatefulWidget {
     super.initState();
-    //getAllContact();
+    setState(() {
+      //contactList = ContactsRepository.getAllContacts()!;
```
Remove code comments

lngyeen commented on lib/ui/screens/contact_list_screen.dart:
```diff
@@ -105,30 +118,34 @@ class _ContactListScreenState extends State<ContactListScreen> {
-                              return Card(
-                                child: contactTile(contactName, phone, index),
-                              );
+                    if (filteredContactList.isNotEmpty)
```
Better if create ValueNotifier<List<Contact>> _filteredContactListNotifier and use ValueListenableBuilder here to rebuild the list.
setState() function will cause whole widget will be rebuild, that's not necessary and losing performance.

lngyeen commented on lib/ui/screens/contact_list_screen.dart:
```diff
@@ -95,6 +99,15 @@ class _ContactListScreenState extends State<ContactListScreen> {
   }
 
+  _onChanged(String value){
+    setState(() {
```
Better if create ValueNotifier<List> _filteredContactListNotifier and use ValueListenableBuilder to rebuild the list.

heupotato committed:
Update contact_list_screen

Remove code comments
Resolve: "ContactsRepository.getAllContacts() should always return List, not optional value."

heupotato committed:
Use valuenotifier instead of setState

heupotato committed:
Update contact_list_screen.dart

heupotato committed:
Update ValueListenableBuilder
